src/systems/a320_systems/src/hydraulic/mod.rs

```diff
@@ -5010,64 +5193,127 @@ impl RudderSystemHydraulicController {
         }
     }
 
-    fn update_rudder(
-        &mut self,
-        green_circuit_available: bool,
-        blue_circuit_available: bool,
-        yellow_circuit_available: bool,
-    ) {
-        let no_hydraulics =
-            !green_circuit_available && !blue_circuit_available && !yellow_circuit_available;
-
-        if self.is_powered && !no_hydraulics {
-            self.set_rudder_position_control(
-                green_circuit_available,
-                blue_circuit_available,
-                yellow_circuit_available,
-            );
+    fn update_rudder_control_state(&mut self) {
+        if self.any_hyd_available() {
+            self.set_rudder_position_control();
         } else {
             self.set_rudder_no_position_control();
         }
     }
 
-    fn update(
+    fn any_hyd_available(&self) -> bool {
+        self.green_press_control_avail
+            || self.blue_press_control_avail
+            || self.yellow_press_control_avail
+    }
+
+    fn update_pressure_availability(
         &mut self,
         green_pressure: &impl SectionPressure,
         blue_pressure: &impl SectionPressure,
         yellow_pressure: &impl SectionPressure,
     ) {
-        self.update_rudder_requested_position();
+        if green_pressure.pressure_downstream_leak_valve().get::<psi>()
+            > Self::MIN_PRESSURE_HI_HYST_FOR_ACTIVE_CONTROL_PSI
+        {
+            self.green_press_control_avail = true;
+        } else if green_pressure.pressure_downstream_leak_valve().get::<psi>()
+            < Self::MIN_PRESSURE_LO_HYST_FOR_ACTIVE_CONTROL_PSI
+        {
+            self.green_press_control_avail = false;
+        }
```

Comment:
Looks like you didn't like my suggestion?

Reply:
It won't compile, if there is no `else` statement then the return type of the `if`-expression will be `()`, since it is possible that no block is evaluated. That will obviously not be possible to assign to the variable. And adding an `else` block would not be a hysteresis anymore (i.e. change the logic)

Reply:
My bad you're right.